### DISCUSSION TITLE
format and location of 'indieweb'   "me" links

### DISCUSSION THREAD

**Discussion by douglasjsmith404** [importantimport/urara#79] [Q&A] [answered]
using urara where do i place the links needed to verify the site as 'me' ... 
i have the integrated apache server running on my mac.    here ->.   [my site](https://douglasjsmith404.com) 
do i need to add a  `.well-known` directory ?
or could i add links in /src/lib/config/general.ts ?
after running `pnpm build` won't that overwrite the links?

**Comment by Interstellar750** [comment]
You can look at the **Head Config** chapter in #31 for instructions on this, but I don’t know the detailed process, so I can’t give you much advice.

Also, @kwaa has [an article](https://kwaa.dev/indiekit) about this, but it's not written in English, you may need to use a translator.

**Comment by douglasjsmith404** [comment]
hey all!  i was able to login to indieweb site by adding the required 'me' links as in:
` - <a href="https://github.com/douglasjsmith404" rel="me">@douglasjsmith404 on Github</a>`
`-  <a rel="me" href="https://douglasjsmith404.github.io/indieweb">IndieWeb/gh-pages</a>`
`-  <a rel="me" href="https://mastodon.social/@douglasjsmith404">Mastodon</a>`

i created an 'indieweb' folder and page in `urara` and included these type of links... you need to have the corresponding links on the different platforms.      Then you can login to different indieweb sites with : 
https://douglasjsmith404.com/indieweb/ 

github required an email verification.
mastodon link was supplied with a verification 'action'

**Comment by Interstellar750** [comment, marked as answer]
When I verify your site's `douglasjsmith404.com/indieweb` directory on the [indielogin.com](https://indielogin.com) website, it requests GitHub verification normally, which should be the 

```md
- <a href="https://github.com/douglasjsmith404" rel="me">@douglasjsmith404 on Github</a>
```

worked.

However, this is only for the `douglasjsmith404.com/indieweb` directory, because other pages in your site do not have `rel="me"` information, for example, you are in [indielogin.com](https://indielogin.com) When authenticating using your site's homepage `douglasjsmith404.com`, you get the error "We couldn't find any way to authenticate you using your website." Is this what you expected?

If you want all pages of your site to support Indielogin login, then you should modify the `/src/lib/config/general.ts` file with the following settings:

```ts
export const head: HeadConfig = {
   custom: ({ dev }) =>
     dev
       ?[]
       :[
           // The HTML tags filled in here will be added to the <head> tag of the website without modification
           // IndieAuth
           '<link rel="authorization_endpoint" href="https://indieauth.com/auth">',
           '<link rel="token_endpoint" href="https://tokens.indieauth.com/token">',
         ],
   me: [
         // It will be formatted as `<link rel="me" href="url">` and added to the <head> tag of the website
         // Fill in the URL that supports Indielogin
         'https://github.com/username',
         'https://twitter.com/username'
       ]
}
```
